Automattic/wp-calypso
# My Home: Fix “Hide this” button isn't usable with keyboard

#### Changes proposed in this Pull Request

* The "Hide this" button on promotional cards isn't usable with a keyboard because the **PopoverMenu** isn't focus on when it shows. To make **PopoverMenu** focus when it shows, it provides **_onShow** method to **Popover** component, but **Popover** component doesn't use it at all. So I make it support this property when **componentDidMount**
* As we need to focus on **PopoverMenu**, **Popover** component has **scheduled focus** to focus on itself to still not focus on **PopoverMenu**. Therefore, set **isFocusEnabled** to false to disable it permanently.
* **suggestion-list** (used by user-mention) also disable **isFocusEnabled** to avoid losing focus on text input and control the keyboard itself, so check this property to avoid focusing on **PopoverMenu**. I think the logic about show and focus might be a little bit strange, any good idea?
* Add **aria-haspopup** and **aria-expanded** to improve a11y.

https://user-images.githubusercontent.com/13596067/126124081-75e18a55-100b-484f-a326-32243cc0233a.mov

<img width="300px" src="https://user-images.githubusercontent.com/13596067/126127201-6a47d8ab-9afe-4d7d-8e69-a22469fbc8c9.png" />

#### Testing instructions



Hide this

* Go to My Home and click the “Hide this” button
* Press “Arrow Up”, “Arrow Down” and “Tab” to see the result
* After closed the menu, press “Tab” again to see it focus on next element or not

User mentions

* Visit http://calypso.localhost:3000/devdocs/blocks/user-mentions
* Press '@' and start typing for user matches
* User suggestions show up and textarea remains in focus, allowing the user to perform keyboard operations as expected.



Related to https://github.com/Automattic/wp-calypso/issues/53617
